### DISCUSSION TITLE
Python example for AuthorEmailPlanner

### DISCUSSION THREAD

**Discussion by dbogdoll** [microsoft/semantic-kernel#6129] [General]
Hi all,

where could I find the Python Example for the page https://learn.microsoft.com/en-us/semantic-kernel/agents/?
Or at least for the AuthorEmailPlanner mentioned there?

Thanks,
Dieter

**Comment by obragaa** [comment]
It seems like you're looking for a Python example for the Planner mentioned on the Microsoft page. Here's a translation of the provided code example:

```python
class AuthorEmailPlanner:
    @KernelFunction
    @Description("Returns the necessary steps for writing an email.")
    async def GenerateRequiredStepsAsync(self, kernel, topic, recipients):
        # Requests the LLM to generate a list of steps to complete the task
        result = await kernel.InvokePromptAsync(f"""
        I'm going to write an email to {recipients} about {topic} on behalf of a user.
        Before I do that, could you succinctly recommend the top 3 steps I should take in a numbered list?
        I want to make sure I don't forget anything that would help my user's email sound more professional.
        """, {
            "topic": topic,
            "recipients": recipients
        })

        # Returns the plan to the agent
        return result.toString()
```

This example demonstrates how to create a Planner in Python called `AuthorEmailPlanner`, which generates the necessary steps for writing an email based on the provided topic and recipients. I hope this helps you with what you're looking for!

**Reply by dbogdoll** [reply]
Hi @obragaa ,

which version of the Python Semantic Kernel Library are you using?

When I use your code I get:
```
ERROR:semantic_kernel.functions.kernel_function:Error occurred while invoking function GenerateRequiredSteps: 'Kernel' object has no attribute 'InvokePromptAsync'
```

as in the version I'm using `semantic-kernel = "^0.9.6b1"` there is no such method.

Also the last line `return result.toString()` looks more like Java or C# as Python. In Python I would exepct something like `str(result)`.

**Reply by dbogdoll** [reply]
Hi @sophialagerkranspandey,

could you please reopen the issue, as it is IMHO still open.

Thanks.
